### ISSUE TITLE
Improve how model.cond works

### ISSUE THREAD

**Issue by grst** [scverse/multi-condition-comparisions#15]
### Description of feature

- [x] The implementation for finding the baseline level for each variable is currently hacky. Need to reach out to the formulatic devs for input. 
- [ ] The function should print out the actual contrast used for diagnostics

**Comment by const-ae** [comment]
Hi Gregor, 

I was looking into getting the default values again and was wondering if the following works or if there was a reason why we dismissed it?

```python
import formulaic
import pandas as pd
import numpy as np

data = pd.DataFrame({"celltype": ['A', 'A', 'B', 'C', 'C', 'C'],
        "condition": ['a', 'b', 'a', 'b', 'a', 'b'],
        "cont": [1, 2, 3, 4, 5, 6]})
formulaic.model_matrix('~ condition', data).model_spec.encoder_state['condition'][1]['categories']
data["condition"] = pd.Categorical(data["condition"], categories = ["b", "a"], dtype = "category")
formulaic.model_matrix('~ condition', data).model_spec.encoder_state['condition'][1]['categories']
```

**Comment by grst** [comment]
The reason was that if you use the syntax described [here](https://matthewwardrop.github.io/formulaic/guides/contrasts/#treatment-aka-dummy), it breaks: 

```python
>>> formulaic.model_matrix('~ C(condition, contr.treatment(base="b"))', data).model_spec.encoder_state['C(condition, contr.treatment(base="b"))'][1]['categories']
['a', 'b']
```

Of course, also something we could decide not to support initially, but it would be nice to have a robust way to achieve this.

**Comment by const-ae** [comment]
Ah yes. But I think it is fine to nonetheless use 'a' as a reference level. I think the reference level should be based on whatever the data contains and not which additional modifications the formula applies.

I think a helpful analogy is to consider a continuous covariate and `formulaic.model_matrix('~ I(cont + 10)', data)`. The formula shifts all values up by `10`, but that doesn't mean that we would move the continuous intercept to `-10`.  Applying the same logic to the categorical covariates means that I think it's fine to just look at the categories.

**Comment by grst** [comment]
But it does change the design matrix accordingly?

![image](https://github.com/scverse/multi-condition-comparisions/assets/7051479/a12804b1-f056-490b-9933-2042362a62cf)

**Comment by grst** [comment]
Also adding here that I think this function should print out the actual contrast vector used (maybe omit zeros). 
It's already a dataframe, so something like

```python
contrasts.T.loc[lambda x: x[0] != 0]
```

could work nicely.

**Comment by grst** [comment]
Closed by #36
Follow up in https://github.com/theislab/pertpy/issues/612
